scrapbook/tests/test_notebooks.py

```diff
@@ -51,10 +51,24 @@ def test_bad_path():
 
 
 def test_bad_ext():
-    with pytest.raises(ValueError):
+    with pytest.raises(Warning):
         Notebook("not/a/valid/extension.py")
 
 
+def test_good_ext_for_url():
+    with pytest.raises(PapermillOptionalDependencyException):
```

Comment:
You shouldn't need this for testing. Do `pip install .[test]` or `pip install .[dev]` (which the test suite does) and you'll install the optional dependencies.

Reply:
I'm stuck on getting this test to pass correctly. In my understanding: 

- requirements-dev.txt contains the papermill dependency
- setup.py then specifies `reqs_dev = read_reqs("requirements-dev.txt") + reqs_all` with the latter containing the azure dependency
- tox.ini sets dependencies to .[dev]

However, tox does not install azure dependencies to it's site-packages env folder. Am I missing something?

Reply:
Oh looks like our setuptools isn't correct and missed adding the particular entrypoints for reqs_dev. We need to add `papermill[all]` added to those dependencies. Our tests don't actually ever reach out to remote interfaces so they don't need the specialized io entrypoint to run. But our dev should include them at a minimum and having the test suite install them checks for obvious import issues. `pip install .[azure]` will pull the dependency you're missing. If you want to fix the reqs_dev as well in this PR we can merge that then, or I'll add it later.

Reply:
Would you care to elaborate a bit? Something along the lines of: `reqs_dev = read_reqs("requirements-dev.txt").extend(reqs_all)`?

Reply:
probably just change the `papermill` dependencies to `papermill[all]` in requirements-dev.txt. I'll make the change independently, sorry it tripped you up